src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt

```diff
@@ -0,0 +1,303 @@
+            page = ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS
+        }
+        while (reindexInformatieobjecten(listParameters)) {
+            listParameters.page++
+        }
+    }
+
+    private fun reindexAllTaken() {
+        val numberOfTasks = flowableTaskService.countOpenTasks()
+        var page = 0
+        while (reindexTaken(page, numberOfTasks)) {
+            page++
+        }
+    }
+
+    private fun reindexZaken(listParameters: ZaakListParameters): Boolean {
+        val zaakResults = zrcClientService.listZaken(listParameters)
+        indexeerDirect(
+            zaakResults.results.map { it.uuid.toString() },
+            ZoekObjectType.ZAAK,
+            performCommit = false
+        )
+        logProgress(
+            ZoekObjectType.ZAAK,
+            (listParameters.page - ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS) * Results.NUM_ITEMS_PER_PAGE +
+                zaakResults.results.size,
+            zaakResults.count.toLong()
+        )
+        return zaakResults.next != null
+    }
+
+    private fun reindexInformatieobjecten(
+        listParameters: EnkelvoudigInformatieobjectListParameters
+    ): Boolean {
+        val enkelvoudigInformatieObjectenResults = drcClientService.listEnkelvoudigInformatieObjecten(listParameters)
+        indexeerDirect(
+            enkelvoudigInformatieObjectenResults.results.map { URIUtil.parseUUIDFromResourceURI(it.url).toString() },
+            ZoekObjectType.DOCUMENT,
+            performCommit = false
+        )
+        logProgress(
+            ZoekObjectType.DOCUMENT,
+            (listParameters.page - ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS) * Results.NUM_ITEMS_PER_PAGE +
+                enkelvoudigInformatieObjectenResults.results.size,
+            enkelvoudigInformatieObjectenResults.count.toLong()
+        )
+        return enkelvoudigInformatieObjectenResults.next != null
+    }
+
+    private fun reindexTaken(page: Int, numberOfTasks: Long): Boolean {
+        val firstResult = page * TAKEN_MAX_RESULTS
+        val tasks = flowableTaskService.listOpenTasks(
+            TaakSortering.CREATIEDATUM,
+            SorteerRichting.DESCENDING,
+            firstResult,
+            TAKEN_MAX_RESULTS
+        )
+        indexeerDirect(
+            tasks.map { it.id },
+            ZoekObjectType.TAAK,
+            performCommit = false
+        )
+        if (tasks.isNotEmpty()) {
+            logProgress(ZoekObjectType.TAAK, firstResult.toLong() + tasks.size, numberOfTasks)
+            return tasks.size == TAKEN_MAX_RESULTS
+        }
+        return false
+    }
+
+    private fun logTypeMessage(objectType: ZoekObjectType, message: String) =
+        LOG.info("[$objectType] $message")
+
+    private fun logProgress(objectType: ZoekObjectType, voortgang: Long, grootte: Long) =
+        logTypeMessage(objectType, "reindexed: $voortgang / $grootte")
+
+    private fun <T> wrapInIndexingException(fn: () -> T): T {
```

Comment:
Bit of a weird construct this I find but let's leave this for now I guess.

Reply:
Tried to get rid of code duplication, but indeed this is not quite standard